File: exercism/scala/bob/src/main/scala/bob.scala
class Bob {
  val Yell = "(?=.*[A-Z].*)[^a-z]+".r
  val Question = """\?$""".r.unanchored
  val Silence = """\s*""".r

  def hey(msg: String):String = msg match {
    case Silence() => "Fine. Be that way!"
    case Yell() => "Whoa, chill out!"
    case Question() => "Sure."
    case _ => "Whatever."
  }
}

File: exercism/elixir/bob/bob.exs
defmodule Teenager do
  def hey(input) do
    cond do
      Regex.match?(~r/^\s*$/,input) -> "Fine. Be that way!"
      String.at(input, -1) == "?" -> "Sure."
      Regex.match?(~r/(?=\w)[^\d]/,input) && String.upcase(input) == input -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end
